### PR TITLE
feat: Sync visiting controllrs to CTS when accepted

### PR DESCRIPTION
**Pull request by AxonC**
Not the finest bit of code, but it does appear to work. I don't have a full live CTS to test against locally, but this matches what data is expected in the production system for a visiting controller.

The logic for the internal IDs is straight from CTS. I dare not simplify it in case it is somehow messed up (there is a refactoring opportunity I can spot straight away with the boolean logic.). 

Also, there are often things not worth unit testing. I firmly put this in that category.

### CODE REVIEW

File: app/Models/Mship/Concerns/HasCTSAccount.php [hunk 1]
```diff
@@ -26,8 +44,33 @@ public function syncToCTS()
 
             $ctsAccount = DB::table("{$ctsDatabase}.members")->where('cid', $this->id)->first();
 
+            // if no CTS account exists, lets sync to create a new one, and not update anything this time round.
             if (! $ctsAccount) {
-                // No user exists. Abort.
+                // for a division member, use the join timestamp of them joining, else use an empty timestamp.
+                // this is how CTS in its wisdom puts this column on visiting controllers.
```
Comment (CalumTowers) [inline app/Models/Mship/Concerns/HasCTSAccount.php:50]: Reword please!

File: app/Models/Mship/Concerns/HasCTSAccount.php [hunk 2]
```diff
@@ -26,8 +44,33 @@ public function syncToCTS()
...
+                $is_visitor = $this->primary_permanent_state->code != 'DIVISION';
+                $joined_div = ! $is_visitor
+                    ? $this->primary_permanent_state->pivot->start_at
+                    : gmdate('Y-m-d H:i:s');
+
+                $newMember = [
+                    'old_rts_id' => 0,
+                    'id' => $this->generateCTSInternalID($this->id),
+                    'cid' => $this->id,
+                    'name' => $this->real_name,
```
Comment (CalumTowers) [inline app/Models/Mship/Concerns/HasCTSAccount.php:60]: Does this honour nicknames?

File: app/Models/Mship/Concerns/HasCTSAccount.php [hunk 3]
```diff
@@ -26,8 +44,33 @@ public function syncToCTS()
 
             $ctsAccount = DB::table("{$ctsDatabase}.members")->where('cid', $this->id)->first();
 
+            // if no CTS account exists, lets sync to create a new one, and not update anything this time round.
             if (! $ctsAccount) {
-                // No user exists. Abort.
+                // for a division member, use the join timestamp of them joining, else use an empty timestamp.
+                // this is how CTS in its wisdom puts this column on visiting controllers.
+                $is_visitor = $this->primary_permanent_state->code != 'DIVISION';
+                $joined_div = ! $is_visitor
```
Comment (CalumTowers) [inline app/Models/Mship/Concerns/HasCTSAccount.php:52]: Double check the live CTS - I feel like this should be null if the member is visiting (as they've never actually joined us). If it's not nullable, I would use the state start_at for the visiting row as the joined at.

**Review by CalumTowers — CHANGES_REQUESTED** [review]
Could you update the variables_with_underscores to be camelCase for consistency? Otherwise seems good to me!

File: app/Models/Mship/Concerns/HasCTSAccount.php (hunk 3 above)
Comment (AxonC) [reply app/Models/Mship/Concerns/HasCTSAccount.php:52]: From what I could find, it is literally this empty type timestamp (gets written as something like 00-00-0000 00:00:00). It's a non-nullable column with no default.

File: app/Models/Mship/Concerns/HasCTSAccount.php (hunk 2 above)
Comment (AxonC) [reply app/Models/Mship/Concerns/HasCTSAccount.php:60]: Probably not, but existing syncing via cron does not from what I've seen (happy to be wrong there)

**Review by atoff — DISMISSED** [review]

File: app/Models/Mship/Concerns/HasCTSAccount.php (hunk 1 above)
Comment (AxonC) [reply app/Models/Mship/Concerns/HasCTSAccount.php]: Done :)

File: app/Models/Mship/Concerns/HasCTSAccount.php (hunk 2 above)
Comment (CalumTowers) [reply app/Models/Mship/Concerns/HasCTSAccount.php:60]: Let's honour nicknames as we do for other services like forum!

Comment (AxonC) [reply app/Models/Mship/Concerns/HasCTSAccount.php]: Done! https://github.com/VATSIM-UK/core/pull/2231/commits/6499524184d03b87e6ba300ad45f727fc9c87018

**Review by CalumTowers — APPROVED** [review]

**Pull request merged**